reports/glides_terminal_departure_accuracy.livemd

```diff
@@ -25,8 +26,503 @@ alias TransitData.GlidesReport
 Kino.nothing()
 ```
 
-## Section
+## Instructions
+
+This notebook provides an implementation of [🚃 Build External Glides Terminal Departure Accuracy Report](https://app.asana.com/0/584764604969369/1206879941322109/f).
+
+The notebook requires AWS credentials for a user with access to the `mbta-gtfs-s3` family of S3 buckets.
+
+<details>
+<summary><kbd><strong>How to add your AWS credentials</strong></kbd></summary>
+<br/>
+
+---
+
+1. Open your Hub. [This link](/hub/personal-hub) should send you there.
+1. Under "Secrets", add two secrets with the following names:
+   - AWS_ACCESS_KEY_ID
+   - AWS_SECRET_ACCESS_KEY
+1. Return to this notebook. Click the 🔒 icon in the left sidebar and toggle on both secrets.
+
+---
+</details>
+
+### Generating a report
+
+Evaluate code cells from top to bottom.
+
+Some cells produce controls that let you adjust how the report runs.
+
+Cells that generate inputs / outputs are marked with a "👉"—you can skip to these, and when you evaluate them, any intermediate cells will automatically evaluate as well.
+
+**Tip:** If you want to generate another report with different settings, simply change the settings, scroll back down to [**Results**](#results), and click "Evaluate" again.
+
+## Setup
+
+```elixir
+env_input =
+  Kino.Input.select("Environment", [
+    {"", "prod"},
+    {"-dev-blue", "dev-blue"},
+    {"-dev-green", "dev-green"},
+    {"-dev", "dev"},
+    {"-sandbox", "sandbox"}
+  ])
+
+today_date =
+  DateTime.utc_now()
+  |> DateTime.shift_zone!("America/New_York")
+  |> DateTime.to_date()
+
+yesterday_date = Date.add(today_date, -1)
+
+# Default setting: Analyze a full service day, starting at 4am yesterday (Eastern) and ending at 4am today (Eastern).
+yesterday_start_time =
+  DateTime.new!(yesterday_date, ~T[04:00:00], "America/New_York")
+  |> DateTime.shift_zone!("Etc/UTC")
+  |> DateTime.to_naive()
+
+today_end_time =
+  DateTime.new!(today_date, ~T[03:59:59], "America/New_York")
+  |> DateTime.shift_zone!("Etc/UTC")
+  |> DateTime.to_naive()
+
+start_date_input = Kino.Input.utc_datetime("Analyze data from...",
+        default: yesterday_start_time,
+        max: DateTime.utc_now()
+      )
```

Comment:
Looking at the Kino API I don't think you can easily do anything about this, so this is more an interesting observation. Because the input is based off the users Timezone for me the defaults go from 2 a.m. to 2 a.m. instead of 4 to 4.
![Screenshot 2024-07-19 at 12 51 39 PM](https://github.com/user-attachments/assets/50ec7744-581b-4442-aafd-b1d8c67b41d9)

But then when bucketed it uses EST so I get results ranging between 4 a.m. and 4 a.m.

It doesn't seem like you can control the TZ of the Kino input, and I don't think it makes sense to display the results in my local timezone, so this is just me whining.

Reply:
Yeah, there's no way to force the input to display times in a particular TZ unfortunately. May I suggest changing your system clock to Eastern Time? 😛